### ISSUE TITLE
Missing "forces new resource" in plan output

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform#20381]
I strongly suspect this may be just a formatting bug, but I didn't have time to dig into it yet, I just don't want to forget about it so I'm creating this issue.

### Terraform Version

0ff04b123271c3bd263f45b1340d2caf5202b3f9

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-west-2"
}

data "aws_ami" "ubuntu" {
  most_recent = true

  filter {
    name   = "name"
    values = ["*ubuntu-xenial-16.04-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  filter {
    name   = "root-device-type"
    values = ["ebs"]
  }

  owners = ["099720109477"] # Canonical
}

resource "aws_key_pair" "radek" {
  key_name   = "radeksimko-teamcity-test"
  public_key = "${file("~/.ssh/id_rsa.pub")}"
}

resource "aws_instance" "web" {
  ami           = "${data.aws_ami.ubuntu.id}"
  instance_type = "t2.medium"
  associate_public_ip_address = true
  # key_name      = "${aws_key_pair.radek.key_name}"
}
```

## How to reproduce

1. Use the config above
1. `terraform apply`
1. Uncomment line with `key_name`
1. `terraform plan`

## v0.11

![screen shot 2019-02-18 at 17 21 56](https://user-images.githubusercontent.com/287584/52967409-e79ba800-33a1-11e9-831e-01001b1b63dd.png)

## v0.12 (currently)

![screen shot 2019-02-18 at 17 23 43](https://user-images.githubusercontent.com/287584/52967445-000bc280-33a2-11e9-9a9a-088069bb25c3.png)

**Comment by apparentlymart** [comment]
Ahh, I think the problem here is that it's an addition rather than an update and so it's going into the "write single value" codepath rather than the "write diff" codepath, and the former doesn't have the checks for whether the change requires replacement. :thinking:

**Comment by GBrawl** [comment]
Looked into it a little bit and that is indeed what is going on, since old.IsNull() will evaluate to true it will go into the writeValue flow instead of writeValueDiff. We could fix this by putting a check below attrS.Sensitive something like this:
```
	if attrS.Sensitive {
		p.buf.WriteString("(sensitive value)")
	} else if p.pathForcesNewResource(path) {
               p.buf.WriteString(p.color.Color(forcesNewResourceCaption))
         }
            else {
		switch {
		case showJustNew:
			p.writeValue(new, action, indent+2)
		default:
			// We show new even if it is null to emphasize the fact
			// that it is being unset, since otherwise it is easy to
			// misunderstand that the value is still set to the old value.
			p.writeValueDiff(old, new, indent+2, path)
		}
	}
```
or we could change the logic of showJustNew

**Comment by GBrawl** [comment]
I created a PR with a fix for this problem, could someone check if this solution is sufficient?

**Comment by apparentlymart** [comment]
This was fixed by #20827.
